app/authorization.service.ts

```diff
@@ -1,36 +1,74 @@
 import {Injectable} from '@angular/core';
-import {Observable} from 'rxjs/Observable'
-import {BehaviorSubject} from 'rxjs/BehaviorSubject'
+import {Observable} from 'rxjs/Observable';
+import {Http, Headers, Response} from '@angular/http';
+import {RequestOptions} from '@angular/http';
+import 'rxjs/add/operator/map';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {User} from './user';
 
 @Injectable()
 
 export class AuthorizationService {
     isLoginSubject: BehaviorSubject<boolean>;
-    currentUser: any;
+    baseURL: string;
+    token: string;
+    user: User;
 
-    constructor() {
-        this.currentUser = JSON.parse(localStorage.getItem('currentUser'));
-        this.isLoginSubject = new BehaviorSubject(!!this.currentUser);
+    constructor(private http: Http) {
+        var currentUser = JSON.parse(localStorage.getItem('currentUser'));
+
+        this.token = currentUser && currentUser.token;
+        this.isLoginSubject = new BehaviorSubject(!!currentUser);
+        this.baseURL = 'http://localhost:3004';
+        this.user = new User();
     }
 
-    login(name: string, password: string): void {
-        this.currentUser = {name, password};
+    login(name: string, password: string): Observable<boolean> {
+        return this.http.post(`${this.baseURL}/auth/login`, {login: name, password: password})
+            .map((res: Response) => {
+                let response = res.json(),
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Body/json

According to the API, response.json() returns promise, so it's unsafe to use this code synchronously. Consider splitting it into two steps.